odbc_fdw.c

```diff
@@ -1403,11 +1383,11 @@ odbcIterateForeignScan(ForeignScanState *node)
         int k;
         bool found;
 
-        #if !defined (TMPHACK)
-        /* Allocate memory for the masks */
+        /* Allocate memory for the masks in a memory context that
+           persists between IterateForeignScan calls */
+        prev_context = MemoryContextSwitchTo(estate->es_query_cxt);
```

Comment:
Here we're using it in its simplest form, since we don't need to create a new memory context, because there's one in the executor state that has the scope we need (query scope). So we simply use it and keep the previous contest in use to restore it when we're done.